pandas/io/tests/test_parsers.py

```diff
@@ -2690,6 +2690,42 @@ def test_uneven_lines_with_usecols(self):
         df = self.read_csv(StringIO(csv), usecols=usecols)
         tm.assert_frame_equal(df, expected)
 
+    def test_read_empty_with_usecols(self):
+        # See gh-12493
+        names = ['Dummy', 'X', 'Dummy_2']
+        usecols = names[1:2] # ['X']
+
+        def check_usecols(stringIO, expected):
+            df = self.read_csv(stringIO, names=names, usecols=usecols)
+            tm.assert_frame_equal(df, expected)
+
+        # first, check to see that the response of
+        # parser when faced with no provided columns
+        # throws the correct error, with or without usecols
+        name = self.__class__.__name__
```

Comment:
That was a vestige from when I had the `klass = CParser if...else StopIteration`.  I'll remove it.